elcometer456_block.py

```diff
@@ -42,13 +58,15 @@ def _read_gage(self):
             self.logger.debug('Waiting for reading')
             try:
                 raw = self._serial.readline()
-            except:
+            except serial.SerialException:
                 if not self._stopping:
-                    self.logger.warning('serial failed read', exc_info=True)
+                    self.logger.info('Closing serial connection')
+                    self._serial.close()
                 continue
             self._serial.write(b"O")
-            read = b''
+            read = 0
```

Comment:
When does this occur? I see in your test, it looks like this happens if the serial read line response is not valid. In that case, you would want to indicate that by either a) not notifying a signal or b) notifying a signal with value: None. Otherwise, it appears that the gage read a value of 0.